examples/cross_decomposition/plot_pcr_vs_pls.py

```diff
@@ -0,0 +1,110 @@
+# The data
+# --------
+#
+# We start by creating a simple dataset with two features. The target `y` is
+# strongly correlated with the second feature, which has a low variance. The
+# first feature has a higher variance but no predictive power on the target.
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+rng = np.random.RandomState(0)
+n_samples = 500
+f0 = rng.normal(scale=10, size=n_samples)
+f1 = rng.normal(scale=1, size=n_samples)
+X = np.c_[f0, f1]
+y = f1 + rng.normal(scale=.1, size=n_samples)
+
+fig, axes = plt.subplots(1, 3, figsize=(10, 3))
+
+axes[0].scatter(f0, f1)
+axes[0].set(ylim=(-10, 10), xlim=(-20, 20), xlabel='f0', ylabel='f1',
+            title='training data')
+axes[1].scatter(f0, y)
+axes[1].set(xlabel='f0', ylabel='y', title='target vs f0')
+axes[2].scatter(f1, y)
+axes[2].set(xlabel='f1', ylabel='y', title='target vs f1')
+plt.tight_layout()
+plt.show()
+
+##############################################################################
+# Projection on one component and predictive power
+# ------------------------------------------------
+#
+# We now create two regressors: PCR and PLS, and for our illustration purposes
+# we set the number of components to 1. For both models, we plot the first
+# component against the target.
+#
+# As expected, the PCA transformation of PCR has dropped the second feature
+# because it has a low variance, which results in the first component having a
+# very low predictive power on the target. On the other hand, the PLS regressor
+# manages to capture the effect of the second feature thanks to its use of
+# target information during the transformation.
+#
+# We also print the R-squared scores of both estimators, which further confirms
+# that PLS is a better alternative than PCR in this case.
+
+from sklearn.model_selection import train_test_split
+from sklearn.pipeline import make_pipeline
+from sklearn.linear_model import LinearRegression
+from sklearn.decomposition import PCA
+from sklearn.cross_decomposition import PLSRegression
+
+X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=rng)
+
+pcr = make_pipeline(PCA(n_components=1), LinearRegression())
+pcr.fit(X_train, y_train)
+pca = pcr.named_steps['pca']  # retrieve the PCA step of the pipeline
+
+pls = PLSRegression(n_components=1)
+pls.fit(X_train, y_train)
+
+fig, axes = plt.subplots(1, 2, figsize=(10, 3))
+axes[0].scatter(pca.transform(X_test), y_test)
+axes[0].set(xlabel='first component', ylabel='y', title='PCR / PCA')
+axes[1].scatter(pls.transform(X_test), y_test)
+axes[1].set(xlabel='first component', ylabel='y', title='PLS')
+plt.tight_layout()
+plt.show()
+print(f"PCR r-squared {pcr.score(X_test, y_test):.3f}")
+print(f"PLS r-squared {pls.score(X_test, y_test):.3f}")
+
+##############################################################################
+# As a final remark, we note that PCR with 2 components performs well, since
+# the second feature wasn't dropped in this case:
```

Comment:
Using 2 components here seems factitious to me, because I know that the target is correlated with only one feature: I know that I'm mixing the info about the two features and the coordinate system where the features are represented.
What about rotating the point cloud X and visualize the result of the PCA decomposition with one component and two component? Just trying to find a way to avoid my mistake.

Reply:
> What about rotating the point cloud X and visualize the result of the PCA decomposition with one component and two component?

Ideally we would do that, but then it would be harder to visualize how `y` is correlated with the direction of least variance

Reply:
I updated the example so that the directions or variance aren't axis-aligned. I think this is what you were referring to?

Reply:
Yes, looks better now, thanks!